aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/Translator.java

```diff
@@ -74,51 +114,131 @@ static DescribeRepositoryRequest translateToReadRequest(final ResourceModel mode
   static ResourceModel translateFromReadResponse(final DescribeRepositoryResponse describeRepositoryResponse) {
     RepositoryDescription repositoryDescription = describeRepositoryResponse.repository();
     return ResourceModel.builder()
-        // TODO add external connections and upstreams to read response
+        // TODO add external connections and policy document
+        // https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-codeartifact/issues/18
         .arn(repositoryDescription.arn())
         .description(repositoryDescription.description())
         .administratorAccount(repositoryDescription.administratorAccount())
         .domainName(repositoryDescription.domainName())
+        .upstreams(translateToUpstreamsFromRepoDescription(describeRepositoryResponse.repository()))
         .domainOwner(repositoryDescription.domainOwner())
         .description(repositoryDescription.description())
         .repositoryName(repositoryDescription.name())
         .build();
   }
 
+  /**
+   * Translates resource model into PutRepositoryPermissionsPolicyRequest
+   * @param resourceModel resource model
+   * @return PutRepositoryPermissionsPolicyRequest
+   */
+  public static PutRepositoryPermissionsPolicyRequest translatePutPermissionsPolicyRequest(
+      ResourceModel resourceModel
+  ) {
+    try {
+      return PutRepositoryPermissionsPolicyRequest.builder()
+          .domain(resourceModel.getDomainName())
+          .repository(resourceModel.getRepositoryName())
+          .domainOwner(resourceModel.getDomainOwner())
+          .policyDocument(MAPPER.writeValueAsString(resourceModel.getPolicyDocument()))
```

Comment:
remind me, I thought this could be either a String or a Map. Not sure how Jackson serializes a string. Does it just return the string directly? or does it wrap it in some json object or something?

Reply:
Per feedback on the Domain resource, we are having this only accept a Map. `resourceModel.getPolicyDocument()` should return a map, which jackson should serialize into a json string.